middlewares/passport.middleware.js

```diff
@@ -1,46 +1,70 @@
-require("dotenv").config();
-const passport = require("passport");
-const LocalStrategy = require("passport-local");
-const users = require("../dao/user.dao");
-const { validatePassword } = require("../utilities/password.util");
+require('dotenv').config();
+const passport = require('passport');
+const LocalStrategy = require('passport-local');
+const GitHubStrategy = require('passport-github2').Strategy;
+const users = require('../dao/user.dao');
+const { validatePassword } = require('../utilities/password.util');
 
 passport.use(
-  new LocalStrategy(
-    { usernameField: "email" },
-    async (username, password, cb) => {
+  new LocalStrategy({ usernameField: 'email' }, async (username, password, cb) => {
+    try {
+      const user = await users.attemptLogin(username);
+      if (!user) {
+        return cb(null, false, {
+          message: 'Incorrect username or password.',
+        });
+      }
+
+      const valid = validatePassword(password, user.password, user.salt);
+
+      if (!valid) {
+        return cb(null, false, {
+          message: 'Incorrect username or password.',
```

Comment:
Something I like to do on the server is to also issue a console.log to record the error and any related information (like a user id). This makes it easier to get the info you need for debugging if a user reports an issue hours or days after it occurred.